frontend/packages/knative-plugin/src/utils/fetch-dynamic-eventsources-utils.ts

```diff
@@ -76,7 +77,14 @@ export const fetchEventSourcesCrd = async () => {
             namespaced: true,
             crd: true,
             color: knativeEventingColor.value,
-          });
+          };
+          const sourceIndex = _.findIndex(accumulator, ['kind', kind]);
+          // added check as some sources has multiple entries with deprecated APIgroups
+          if (sourceIndex === -1) {
+            accumulator.push(sourceModel);
+          } else if (!labels?.['eventing.knative.dev/deprecated']) {
```

Comment:
Along with lables existence, can we also check for value of label as well here?. If yes, then we can also have this label in const so that way we can reuse it.